File: javascript/fibonacci.js
function fibonacci(num) {
  let fibArr = [0, 1]
  if (num >= 2) {
    for (i = 2; i <= num; i++) {
      let newNum = fibArr[i-1] + fibArr[i-2]
      fibArr.push(newNum)
    }
  }
  return fibArr[num];
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

// initialize a starting array with the first two elements
// a for loop creates all the new elements 
// access the element at the index number

File: ruby/fibonacci.rb
def fibonacci(num)
  fib_arr = [0, 1]
  if num >=2 
    for i in 2..num 
      new_num = fib_arr[i-1] + fib_arr[i-2]
      fib_arr << new_num
    end
  end
  return fib_arr[num]
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
